src/codeu/chat/server/View.java

```diff
@@ -78,6 +78,81 @@ public View(Model model) {
   @Override
   public Message findMessage(Uuid id) { return model.messageById().first(id); }
 
+  @Override
+  public Collection<String> userStatusUpdate(String name, Uuid owner) {
+
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+    Collection<String> contributions = new ArrayList<String>();
+
+    final User foundOwner = model.userById().first(owner);
+    final User foundUser = model.userByText().first(name);
+
+    if(foundUser != null) {
+      if(foundOwner.UserSet.contains(foundUser.id)) {
+        final Time lastUpdate = foundOwner.UserUpdateMap.get(foundUser.id);
+        for(ConversationPayload conversationPayload : allConversations) {
+          Message currentMessage = model.messageById().first(conversationPayload.firstMessage);
+          boolean foundMessage = false;
+          while(currentMessage != null && foundMessage == false) {
+            if(lastUpdate.compareTo(currentMessage.creation) < 0 && currentMessage.author.equals(foundUser.id)) {
+              ConversationHeader conversationContribution = model.conversationById().first(conversationPayload.id);
+              contributions.add(conversationContribution.title);
+              LOG.info("Conversation added to list.");
+              foundMessage = true;
```

Comment:
status-update-u <name> is meant to give the user that requests it a list of conversations that the person they specify has recently contributed to. I just kind of assumed that if at least one message from the specified person is in the conversation and had a creation time AFTER the last time a status update was called on them, they contributed to the conversation.